src/workload_api.rs

```diff
@@ -120,3 +122,126 @@ pub async fn fetch_and_write_x509_svid(
 
     Ok(())
 }
+
+/// Creates an X509Source for continuous X.509 certificate watching.
+///
+/// This function creates an X509Source that automatically watches for certificate updates
+/// from the SPIRE agent. It includes retry logic with exponential backoff for initial connection.
+///
+/// # Arguments
+///
+/// * `agent_address` - The address of the SPIRE agent (e.g., "unix:///tmp/agent.sock")
+///
+/// # Returns
+///
+/// Returns `Ok(Arc<X509Source>)` if successful, or an error if connection fails after retries.
+pub async fn create_x509_source(agent_address: &str) -> Result<Arc<X509Source>> {
+    // Convert agent_address to the format expected by spiffe crate
+    let spiffe_path = if agent_address.starts_with("unix://") {
+        let socket_path = agent_address
+            .strip_prefix("unix://")
+            .ok_or_else(|| anyhow::anyhow!("Invalid unix socket address: {agent_address}"))?;
+        // Use unix: prefix (not unix://) for spiffe crate
+        format!("unix:{socket_path}")
+    } else {
+        agent_address.to_string()
+    };
+
+    // Create X509Source with retries (workload may need time to attest)
+    let mut last_error_msg = None;
+    for attempt in 1..=10 {
```

Comment:
@cursoragent this control flow's indentation is too deep. this is bad code smell.

additionally
1. define the retry attempt max in a const value